src/SQLiteInterface.hs

```diff
@@ -1,23 +1,22 @@
 {-# LANGUAGE OverloadedStrings, GADTs #-}
 module SQLiteInterface where
 
-import Data.Maybe (listToMaybe, isJust)
-import Data.ByteString (ByteString)
-import qualified Data.ByteString as B
-import qualified Data.ByteString.Base16 as Base16
-import Data.Serialize (decode, encode)
-
-import qualified Database.SQLite.Simple as SQL
-
+import Control.Concurrent.STM (TVar, STM, atomically, retry)
 import Control.Monad (when)
+import Control.Monad.Operational (ProgramViewT (..), view)
 import Control.Monad.Trans (lift)
 import Control.Monad.Trans.Maybe (MaybeT (..))
-import Control.Concurrent.STM
-import Control.Monad.Operational (ProgramViewT (..), view)
-
 import Crypto.Hash.SHA1 (hash)
+import Data.ByteString (ByteString)
+import Data.Maybe (isJust, listToMaybe )
+import Data.Serialize (decode, encode)
+import StoredData (Data (..), ListNode (..), StoreInstr (..), StoreOp)
```

Comment:
Do we also want to be explicit about fields we import? At the moment this breaks are rule of every should be grepable in the file you are currently in.

Also, a style issue: at the moment we don't but a space between the type name and the opening parenthesis. I think we should stick to this style, mainly because it seems to be more standard, judging by popular libraries that I've looked at.